### PR TITLE
OpQueue: Remove goroutine

### PR DESCRIPTION
**Pull request by mattayes**
# Changes

* Remove the goroutine from OpQueue and use a closed status instead.
* Upgrade dependencies.
* Miscellaneous cleanup.

### CODE REVIEW

**Review by vitaminmoo — APPROVED** [review]

File: opqueue.go [hunk 1]
```diff
@@ -29,49 +29,41 @@ var (
...
+	cond    sync.Cond
 	depth   int
 	width   int
 	q       *list.List
 	entries map[ID]*OpSet
+	closed  bool
 }
 
 // NewOpQueue create a new OpQueue.
 func NewOpQueue(depth, width int) *OpQueue {
```
Comment (ajroetker) [inline opqueue.go:42]: optionally NewOpQueue could take a context, and we could pass in the actor context instead of background

**Review by ajroetker — APPROVED** [review]

File: opqueue.go [hunk 2]
```diff
@@ -111,11 +106,14 @@ func (q *OpQueue) Enqueue(id ID, op *Op) error {
...
-	} else if len(set.Ops()) >= q.width {
+		return nil
+	}
+	if len(set.Ops()) >= q.width {
 		return ErrQueueSaturatedWidth
-	} else {
-		set.append(op)
 	}
+
+	// Q (2023-11) (mh): Why don't we signal here?
```
Comment (mattayes) [inline opqueue.go:115]: No need to signal: we only need to signal when we add an entry to the map/linked list.

```suggestion
```

File: opqueue.go [hunk 3]
```diff
@@ -126,19 +123,17 @@ func (q *OpQueue) Enqueue(id ID, op *Op) error {
 // If the OpQueue is closed, then Dequeue will return false
 // for the second parameter.
 func (q *OpQueue) Dequeue() (*OpSet, bool) {
-	q.cond.L.Lock()
```
Comment (epsniff) [inline opqueue.go:129]: This is a pedantic change that frankly makes it a little less clear that one is using the same lock as the conditional.

Comment (epsniff) [reply opqueue.go:129]: These changes should be a separate PR, adding cognitive load to a review that is core to how we process data isn't a smart idea.   We need to keep these kinds of PRs as simple as possible.

**Review by epsniff — APPROVED** [review]

**Comment by epsniff** [comment]
Yeah, getting rid of the Go routine is a nice change.  Thanks.

**Pull request merged**